### PR TITLE
Upgrade govuk_design_system_formbuilder to v2.0.0

### PR DESCRIPTION
**Pull request by duncanjbrown**
## Context

Supersedes #3000

### CODE REVIEW

**Comment by fofr** [comment]
Be wary of any forms without an explicit legend, this new version auto generates one and the generated version `looks_like_it_was_generated`.

This caught us out in a few places on https://github.com/DFE-Digital/get-help-with-tech/pull/556

Screen Shot 2020-09-29 at 11 27 31

**Comment by duncanjbrown** [comment]
Thank you @fofr — will check

**Comment by tvararu** [comment]
@duncanjbrown I will review once you do!

**Comment by duncanjbrown** [comment]
Reviewed and updated. I used @peteryates's regex from https://github.com/DFE-Digital/get-help-with-tech/pull/556#issuecomment-700204483 to try and find places in the code where we had missing legends. I also searched for instances of literal `<legend>` tags in the codebase, as that often indicated a non-formbuilder-generated legend. Last commit's changes reflect what I found.

I'm almost certain I missed something, so before approving this PR, we should go through the entire candidate form and check it by eye. I'm confident the provider side is ok as there are only a couple of forms (offers, notes, users/orgs) to deal with there. But that means, of course, we should check that too.

**Comment by duncanjbrown** [comment]
@tvararu could I take you up on your kind offer to review this via the front end? I can do the provider side

File: app/views/candidate_interface/equality_and_diversity/edit_disabilities.html.erb [hunk 1]
```diff
@@ -6,7 +6,7 @@
     <%= form_with model: @disabilities, url: candidate_interface_update_equality_and_diversity_disabilities_path, method: :post do |f| %>
       <%= f.govuk_error_summary %>
 
-      <%= f.govuk_check_boxes_fieldset :disabilities, legend: { size: 'xl', text: equality_and_diversity_caption + 'Please select all that apply to you' } do %>
+      <%= f.govuk_check_boxes_fieldset :disabilities, multiple: false, legend: { size: 'xl', text: equality_and_diversity_caption + 'Please select all that apply to you' } do %>
```
Comment (peteryates) [inline app/views/candidate_interface/equality_and_diversity/edit_disabilities.html.erb:9]: Just a note on this and beyond the scope of this PR, but `govuk_check_boxes_fieldset` [now takes a `caption` argument](https://govuk-form-builder.netlify.app/introduction/labels-captions-hints-and-legends/#a-label-with-a-caption), which will make this a little easier on the eye.

**Review by tvararu — APPROVED** [review]
```
      .......::::::::::::)..           .......................(::::::........
      .:::::;;;;;;;;):::::::.... .           .......:::::::::::::<......
          <  >>>                   ,.
  .::..  ;   I;L\  /L\.  ..::..   /iL.           |         ..::::::::::::..
        ;    II;L\/LLLL;         / I;L\    \     |     / /\_
             II;..LLLLLL\    _._/ I;:.L\     \   |   / _/J; \
      :     IIIIi;..LLLLL\__/   IIII:..L\____  \###/  /JJI:  \
    ,;     ILIi;;;:...:LLL;\      IIIII;.LLLLL\#####/JJ II;   \
   ;     I LLii;;;.:.. :LLL;\     III;;;::LLLLL\###/JJ IIII;   \_.
  :     IIILiii;;::.... :LLL;|      ;;I;;::.:LLLLLL:;IJ IIIII;:   \__.
           IIIII IIii;;::;..;\          ;;:::...LLLL;IJIII;;    :::   \
:    ;    IIIIIIIIIii;;::.;..      _==|      ;..  :;IJIII;:::    ::    \
    ;    ::::::::::::;;::..;  _==|   )__)  |                            \
 '  '"  "  ""'""'""  ""'"  '"    )_)  )___) ))  ""''"   ""'"  "'" "'"'"'  "'""
        '""   ""^^       ^~   )___) )____))_)   ~~         ""^^^""  '  "  "~"
' ^^            ^        _    )____)_____))__)\      ~^~~^           ^^"
     '^^          ^~      \---__|____/|___|___-\\--        "~"~         "~"
   ''    '^          ~"~   \   oo oo oo oo     /      ~"      '~       ""~"
        ____   ^^^"~   ~~^^^^^^^^^^^^^^^^^^^^^^    ^~^            ^~^^^
      /  o   \     ""'"  __          __ "'"''     '   ''~     ~""~"`    ""''
    < ____     \"'"    /    \   "' /    \       _          _    "~    _
          |     |     |  __  |    |  __  |    /   \      /   \       / |
    '''   |_____|  '  |__||__| '' |__||__| ' |_____| '' |_____| ""  /_/
         '''     ~^^^^      '"""^^'''''^        '''''''""        '''''^^""
   ''          '^^           ~^^~          ~^ '      ~~      '  ^   ^^^^^^^^
```

**Comment by duncanjbrown** [comment]
Made it! Thanks @fofr @tvararu @paulrobertlloyd @peteryates

**Pull request merged**